charmhelpers/contrib/openstack/ha/utils.py

```diff
@@ -56,6 +57,8 @@
     get_hacluster_config
 )
 
+from charmhelpers.contrib.templating import jinja
```

Comment:
This failed when I tested the change because that module is not included in charm-keystone.

Reply:
Thanks, I'll add it into `charm-helpers-hooks.yaml` file. (I copied it manually)

Reply:
You could make this an optional import by adding it into the `render_grafana_dashboard(...)` function.  That way it won't break on (classic) charms that don't use the functionality, and charms that call `render_grafana_dashboard(...)` (if they are classic) would need their `charm-helpers-hooks.yaml` file updated anyway.

Reply:
What do you mean by "optional"? What do you mean by "optional"? Move the import into function or used something like this?
```python
if is_charm_classsic(): 
  from ... import jinja 
esle:
  ...
```

Reply:
Essentially:

```python
def render_grafana_dashboard(...):
    from ... import jinja
    ...
```

i.e. the module is only attempted to be imported if the render_grafana_dashboard() function is actually used.  This means that if a charm doesn't call the function it doesn't need the module synced (as a classic charm).

Reply:
Ok, that's how I understood it at the beginning, but I wasn't sure.